### ISSUE TITLE
.NET Standard Support

### ISSUE THREAD

**Issue by dansiegel** [xamarin/Xamarin.Auth#194]
### Feature description

With the transition to .NET Standard it would be great if Xamarin.Auth were updated to support .NET Standard. This would also eliminate the need for the BCL dependency which results in the need to add an app.config to fix the bindings in a Xamarin Project that is consuming other .NET Standard libraries or is a .NET Standard library itself.

**Comment by DanielMcAssey** [comment]
Any update to this?

**Comment by morrisandy** [comment]
bump!

**Comment by dpmcgarry** [comment]
+1

**Comment by bartvdbilt** [comment]
How to build Xamarin Auth for .NET standard ?

**Comment by Amenti** [comment]
I would love some update on this. It would make working with Xamarin.Auth a great deal easier.

**Comment by moljac** [comment]
@Amenti 

>  It would make working with Xamarin.Auth a great deal easier.

Adding .NET Standard support made my life hell. 1.5.0.4 is waiting for almost 2 months to be published. It is easier for 1 (OK few users), but not all.

@bartvdbilt 

> How to build Xamarin Auth for .NET standard ?

Cake script in the root. targets = {lib, nuget} compile on my 2 macs and 2 different VMs and one PC, but not on our CI bots. Investigating those issues. Currently 1.6 targeted.

**Comment by moljac** [comment]
@bartvdbilt 

NOTE: you'll need to have parallel VS2015 and VS2017 installations. VS2015 cannot handle .NET standard projects and VS2017 WinRT (Windows [Store] and Windows Phone) and Windows Phone Silverlight. I had to split solutions again because of that.

Every update to VS2017 will most likely render VS2015 installation unusable. The workaround (tried everything) is to uninstall VS2015 and reinstall (reboot is optional, but I do it). No need to install SDKs like UWP, WP, WinRT (both Windows (Store) and Windows Phone).

Bottom line - prepare for heavy lifting and reading the books during installations. I do it weekly. 

Mac builds are OK. With VS4Mac .NET Standard projects are supported (not sure if Alpha channel is needed). Though I did not build steps for .NET standard on Mac - it was not supported till recently

**Comment by dansiegel** [comment]
I noticed there is now a netstandard project. I might ask why on earth the decision was made to go with 1.6. That seems like a really poor choice IMO. 1.4 is the highest you can really go and maintain support with UWP, and 2.0 support for UWP while close isn't quite there AFAIK. Besides if you have an existing PCL you can usually get away with 1.0 or 1.1.

**Comment by Amenti** [comment]
@moljac I can imagine the pain. Thanks for your hard work!

**Comment by moljac** [comment]
@dansiegel 

> give them a finger and they'll take an arm

I won the bet. I said as soon as .NET Standard support version war will start. I'm no .NET Standard expert (did not have time to learn more about it, because of Xamarin.Auth), so 1.6 was the lowest version w/o changes in code. 

Did analysis with Portability Analyser:

https://marketplace.visualstudio.com/items?itemName=ConnieYau.NETPortabilityAnalyzer

and 

https://icanhasdot.net/

> Besides if you have an existing PCL you can usually get away with 1.0 or 1.1.

I was not.

Off the record - I plan to rewrite (refactor) most of the library, so it will be 1.1 or even 1.0

The plan is to rewrite HTTP 

*   with HttpWebRequest and HttpWebResponse for 1.0 and

    http://packagesearch.azurewebsites.net/?q=HttpWebRequest

    and 

*   HttpClient for 1.1+ (this would enable use of platform implementations)

    http://packagesearch.azurewebsites.net/?q=HttpClient

**Comment by dansiegel** [comment]
If you're doing a rewrite I might suggest updating to multitarget and use a single csproj. You can use the [MSBuildSdkExtras](/onovotny/MSBuildSdkExtras) from @onovotny. Feel free to ping me on Xamarin Chat if you have any questions, I've done this sort of thing on several libraries.

**Comment by moljac** [comment]
I'm doing partial rewrite/refactoring. I added several utility classes to encapsulate OAuth concepts, so I can add unit tests for querys, parameter checks, state generation, grant type deduction etc...

I'm a bit afraid to do huuge changes because of backward compatibility and the team size of 1.

Thanks for the link I will read about it and ping you if something is not clear for me.

**Comment by hlubovac** [comment]
It's odd that `<TargetFramework>netstandard2.0</TargetFramework>` is being promoted throughout, while components like this are out of date. Imagine how those (myself included) just beginning to dive into this can get confused.

I'm in this situation:
![image](https://user-images.githubusercontent.com/3127736/34398361-9165eb44-eb3b-11e7-9f91-1eca79153f6b.png)

Can anyone can suggest how to make this working, please?

**Comment by VadyaVL** [comment]
Hello everybody.
Someone can say when will Net Standard 2 support?

**Comment by leticiagheno** [comment]
Hey! 
Any news?

**Comment by garymedina** [comment]
Bump! 

Much needed update please!

**Comment by lennondorneles** [comment]
Hello everybody.
Any news? Really is a very important update for everyone.

**Comment by pricrmartins** [comment]
+1

**Comment by dfoulk** [comment]
+1

**Comment by VadyaVL** [comment]
Hello everybody. As a solution, we can only use this library in Android and iOS apps.
You need to use it PageRenderer with your OAuth logic.
The code is usually bigger. But as a temporary solution will be useful.

**Comment by jdtrepanier** [comment]
+1

**Comment by dfoulk** [comment]
Thanks @VadyaVL for the tip!

**Comment by orionstudt** [comment]
Bump! Would love an update on this!

**Comment by ruigaspar** [comment]
The new preview version "1.6.0-alpha03" works for me!
Thank you @moljac

**Comment by kevin-granados** [comment]
@ruigaspar  preview version of xamarin.auth works for you on a cross platfrom project? really want this to work. so i can login to twitter on my mobile app. any new?

**Comment by ruigaspar** [comment]
@kevin-granados I don't know, I'm only using for store the user login data...

**Comment by PaulVrugt** [comment]
this can be closed i guess with the release of 1.6?

**Comment by moljac** [comment]
Closing .NET Standard 1.0 support is added (as reference assembly)
netstandard 1.6 is real port of the PCL, but for bait-n-switch it is not needed!

Thanks and my apology for making you wait. Believe me I would wish more than you to have this support out sooner.
